src/AppBundle/Controller/PublicController.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Delivery;
+use AppBundle\Entity\DeliveryOrder;
+use AppBundle\Entity\DeliveryOrderItem;
+use AppBundle\Entity\StripePayment;
+use AppBundle\Form\StripePaymentType;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Stripe;
+use Sylius\Component\Payment\PaymentTransitions;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * @Route("/{_locale}/pub", requirements={ "_locale": "%locale_regex%" })
+ */
+class PublicController extends Controller
+{
+    /**
+     * @Route("/d/{number}", name="public_delivery")
```

Comment:
`uuid` is better maybe?

Reply:
When an `OrderInterface` goes from `state=cart` to `state=new`, it uses the default [`OrderNumberAssigner`](https://github.com/Sylius/Sylius/blob/f2042d449c5ae3b68df145910cb42fae6c90d254/src/Sylius/Bundle/OrderBundle/NumberAssigner/OrderNumberAssigner.php) class. 
I didn't change anything to the defaults, but for sure it can be overriden, and the `order.number` field is a `VARCHAR(255)`. It actually generates _strings_ like `000000001`.